harpy/_validations.py

```
@@ -257,17 +257,32 @@ def validate_popsamples(infiles, popfile, quiet):
+                sample, segment_id = line.rstrip().split()
+                code_letters.add(segment_id[0])
+                if segment_id in segment_ids:
+                    print_error("An ID segment must only be associated with a single sample.")
+                    print_solution_with_culprits(
+                        "A barcode segment can only be associated with a single sample. For example, [green]C05[/green] cannot not be the segment that identifies both [blue]sample_01[/blue] and [blue]sample_2[/blue].",
+                        "The segment triggering this error is:"
+                        )
+                    click.echo(segment_id)
+                    sys.exit(1)
+                else:
+                    segment_id.add(seg_id)
```

Ruff (0.8.2)

Undefined name `seg_id`

(F821)